File: src/main/java/de/tectoast/emolga/logging/CorrectColorsConverter.java
package de.tectoast.emolga.logging;

import ch.qos.logback.classic.Level;
import ch.qos.logback.classic.spi.ILoggingEvent;
import ch.qos.logback.core.pattern.color.ForegroundCompositeConverterBase;

import static ch.qos.logback.core.pattern.color.ANSIConstants.*;

@SuppressWarnings("SameParameterValue")
public class CorrectColorsConverter extends ForegroundCompositeConverterBase<ILoggingEvent> {
    @Override
    protected String getForegroundColorCode(ILoggingEvent e) {
        if (hasMarker("important", e)) return CYAN_FG;
        return switch (e.getLevel().toInt()) {
            case Level.ERROR_INT -> BOLD + RED_FG;
            case Level.WARN_INT -> YELLOW_FG;
            default -> DEFAULT_FG;
        };
    }

    private boolean hasMarker(String marker, ILoggingEvent e) {
        return e.getMarkerList() != null && e.getMarkerList().stream().anyMatch(m -> m.getName().equals(marker));
    }
}

File: src/main/kotlin/de/tectoast/emolga/logging/CorrectColorsConverter.kt
package de.tectoast.emolga.logging

import ch.qos.logback.classic.Level
import ch.qos.logback.classic.spi.ILoggingEvent
import ch.qos.logback.core.pattern.color.ANSIConstants
import ch.qos.logback.core.pattern.color.ForegroundCompositeConverterBase

class CorrectColorsConverter : ForegroundCompositeConverterBase<ILoggingEvent>() {
    override fun getForegroundColorCode(e: ILoggingEvent): String {
        return if (hasMarker(e = e)) ANSIConstants.CYAN_FG else when (e.level.toInt()) {
            Level.ERROR_INT -> ANSIConstants.BOLD + ANSIConstants.RED_FG
            Level.WARN_INT -> ANSIConstants.YELLOW_FG
            else -> ANSIConstants.DEFAULT_FG
        }
    }

    companion object {
        private fun hasMarker(marker: String = "important", e: ILoggingEvent): Boolean {
            return e.markerList != null && e.markerList.any { it.name == marker }
        }
    }
}